test/unit/au/org/emii/gogoduck/job/JobExecutorServiceSpec.groovy

```diff
@@ -9,35 +10,46 @@ import au.org.emii.gogoduck.test.TestHelper
 class JobExecutorServiceSpec extends Specification {
 
     def job
+    def jobStoreService
     def notificationService
-    def triggerNowCalled
 
     def setup() {
         job = TestHelper.createJob()
         job.uuid = '1234'
+        jobStoreService = Mock(JobStoreService)
         notificationService = Mock(NotificationService)
-        service.notificationService = notificationService
 
-        triggerNowCalled = false
-        JobExecutorJob.metaClass.static.triggerNow = {
-            triggerNowCalled = (it == [job: job])
-        }
+        service.jobStoreService = jobStoreService
+        service.notificationService = notificationService
     }
 
-    def "run sends 'job registered' notification"() {
+    def "register sends 'job registered' notification"() {
         when:
-        service.run(job)
+        service.register(job)
 
         then:
         1 * notificationService.sendJobRegisteredNotification(job)
     }
 
-    def "triggers job executor job"() {
+    def "register saves job"() {
         when:
-
-        service.run(job)
+        service.register(job)
 
         then:
-        triggerNowCalled
+        1 * jobStoreService.save(job)
     }
+
+    // TODO: cannot get this (or something like it) to work.
+    // Leaving it here for now in case someone else wants to have a try, otherwise,
+    // I'll make sure it's gone before this current parcel of work is complete.
+    // def "register queues job"() {
+    //     given:
+    //     JobExecutorService.JOB_QUEUE = Mock(BlockingQueue)
```

Comment:
I've tried a lot of things - nothing worked.  But I didn't spend too long on it - didn't want to hold things up.  This is fundamental though to how ggd queues and processes things though, so I *do* want to come back to it.